textacy/representations/network.py

```diff
@@ -0,0 +1,129 @@
+          are counted together (see :func:`normalized_str() <textacy.spacy_utils.normalized_str>`)
+    """
+    if window_width < 2:
+        raise ValueError('Window width must be >= 2')
+
+    # when len(terms) < window_width, cytoolz throws a StopIteration error
+    window_width = min(window_width, len(terms))
+    if isinstance(terms[0], str):
+        windows = itertoolz.sliding_window(window_width, terms)
+    elif isinstance(terms[0], spacy_token):
+        windows = ((normalized_str(tok) for tok in window)
+                   for window in itertoolz.sliding_window(window_width, terms))
+    else:
+        msg = 'Input terms must be strings or spacy Tokens, not {}.'.format(type(terms[0]))
+        raise TypeError(msg)
+
+    graph = nx.Graph()
+
+    if edge_weighting == 'cooc_freq':
+        cooc_mat = collections.defaultdict(lambda: collections.defaultdict(int))
+        for window in windows:
+            for w1, w2 in itertools.combinations(sorted(window), 2):
+                cooc_mat[w1][w2] += 1
+        graph.add_edges_from(
+            (w1, w2, {'weight': cooc_mat[w1][w2]})
+            for w1, w2s in cooc_mat.items() for w2 in w2s)
+
+    elif edge_weighting == 'binary':
+        graph.add_edges_from(
+            w1_w2 for window in windows
+            for w1_w2 in itertools.combinations(window, 2))
+
+    return graph
+
+
+def sents_to_semantic_network(sents,
+                              edge_weighting='cosine'):
+    """
+    Convert a list of sentences into a semantic network, where each sentence is
+    represented by a node with edges linking it to other sentences weighted by
+    the (cosine or jaccard) similarity of their constituent words.
+
+    Args:
+        sents (list(str) or list(:class:`spacy.Span`))
+        edge_weighting (str {'cosine', 'jaccard'}, optional): similarity metric
+            to use for weighting edges between sentences; if 'cosine', use the
+            cosine similarity between sentences represented as tf-idf word vectors;
+            if 'jaccard', use the set intersection divided by the set union of
+            all words in a given sentence pair
+
+    Returns:
+        :class:`networkx.Graph`: nodes are the integer indexes of the sentences
+            in the input ``sents`` list, *not* the actual text of the sentences!
+
+    Notes:
+        * If passing sentences as strings, be sure to filter out stopwords, punctuation,
+          certain parts of speech, etc. beforehand
+        * Consider normalizing the strings so that like terms are counted together
+          (see :func:`normalized_str() <textacy.spacy_utils.normalized_str>`)
+    """
+    n_sents = len(sents)
+    if isinstance(sents[0], str):
+        pass
+    elif isinstance(sents[0], spacy_span):
+        sents = [' '.join(normalized_str(tok) for tok in
+                          extract.words(sent, filter_stops=True, filter_punct=True, filter_nums=False))
```

Comment:
I don't have a good idea how to cleanly change this, but hard coding in these options (while these are reasonable defaults) might be too restrictive.  Perhaps someone would want to filter out numbers, for instance. If someone wanted to do this, they'd have to send in the sentences already filtered.  Do you want that behavior?